### PR TITLE
Increase coverage for api and database

### PR DESCRIPTION
**Pull request by Lynoxe**
I have 100% coverage on my scripts.

### CODE REVIEW

**Review by L-roro — APPROVED** [review]
First of all, congratulations for such an impressive test coverage on so much code, it certainly looks like this took an enormous amount of effort.

One thing I do not fully understand is the reasoning behind the runTests/runBlocking calls in most tests. I would appreciate it if you could explain to me the reasoning behind these calls, for it is not intuitive at first glance. From what I gathered in google, they are meant to test coroutines. The thing is that it looks that for this purpose, runTests completly should outshine runBlocking according to [this documentation](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/run-test.html). Is there a reason for keeping both types of calls? Moreover, if I am not mistaken, when using mock repositories, there is no need for using coroutines, since the mocks you created return values "Instantly", so why use these methods on calls to mock repositories? I feel like I am missing the answer to this. If it is not something simple that is just going over my head, maybe we should add some comments to explain it in the code.

Most classes in data/db still do not have any documentation. I believe this has already been mentioned in other pull requests as comments, but I am putting it here once again as the classes that are being modified in this pull request still do not get their documentation added. I am aware that in some cases it gets somewhat ridiculous (specially all the 'entity' classes meant to only store information for the db), but I still believe it would be helpful for people that are not familiar with the code. Maybe we could create a specific task for the next sprint to add all this missing documentation.

I'll approve the pull request despite this, in case you do not have time to modify it before Friday. But do tell me if you agree about creating the mentioned tasks for next sprint.

**Comment by Lynoxe** [comment]
> First of all, congratulations for such an impressive test coverage on so much code, it certainly looks like this took an enormous amount of effort.
> 
> One thing I do not fully understand is the reasoning behind the runTests/runBlocking calls in most tests. I would appreciate it if you could explain to me the reasoning behind these calls, for it is not intuitive at first glance. From what I gathered in google, they are meant to test coroutines. The thing is that it looks that for this purpose, runTests completly should outshine runBlocking according to [this documentation](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/run-test.html). Is there a reason for keeping both types of calls? Moreover, if I am not mistaken, when using mock repositories, there is no need for using coroutines, since the mocks you created return values "Instantly", so why use these methods on calls to mock repositories? I feel like I am missing the answer to this. If it is not something simple that is just going over my head, maybe we should add some comments to explain it in the code.
> 
> Most classes in data/db still do not have any documentation. I believe this has already been mentioned in other pull requests as comments, but I am putting it here once again as the classes that are being modified in this pull request still do not get their documentation added. I am aware that in some cases it gets somewhat ridiculous (specially all the 'entity' classes meant to only store information for the db), but I still believe it would be helpful for people that are not familiar with the code. Maybe we could create a specific task for the next sprint to add all this missing documentation.
> 
> I'll approve the pull request despite this, in case you do not have time to modify it before Friday. But do tell me if you agree about creating the mentioned tasks for next sprint.

Thank you!

- You're right about the `runTest` vs `runBlocking`: I should only use `runTest`. A dependency was missing in the gradle so it wasn't proposed and I didn't think about it but I've already replaced them in a commit. 
- Regarding why I used the coroutine scopes for the mock datasources in `LibraryRepositoryImplTest.kt` for example (if that's what you meant), it's because the top-level interface define the methods as `suspend`. So even if actually it turns out that the mock implementation returns directly, the signature stays `suspend` so it needs to be called inside a coroutine scope.
- I'll add a task for next sprint to add more documentation on my classes. I'd rather focus on other functionalities for this sprint.

File: data/src/main/java/com/github/leuludyha/data/db/LibraryDao.kt
[the remark this answers is not in the document]
Comment (Lynoxe) [reply data/src/main/java/com/github/leuludyha/data/db/LibraryDao.kt:1]: Task for next sprint

File: data/src/main/java/com/github/leuludyha/data/db/LibraryDao.kt [hunk 1]
```diff
@@ -92,6 +92,9 @@ interface LibraryDao {
     @Query("SELECT * FROM editions WHERE editionId LIKE :editionId")
     fun getEdition(editionId: String): Flow<EditionEntity>
 
+    @Query("SELECT * FROM editions WHERE (isbn10 LIKE :isbn OR isbn13 LIKE :isbn)")
+    fun getEditionByISBN(isbn: String): Flow<EditionEntity>
+
```
[the remark this answers is not in the document]
Comment (Lynoxe) [reply data/src/main/java/com/github/leuludyha/data/db/LibraryDao.kt:97]: Task for next sprint too

File: data/src/main/java/com/github/leuludyha/data/paging/SearchPagingSource.kt
[the remark this answers is not in the document]
Comment (Lynoxe) [reply data/src/main/java/com/github/leuludyha/data/paging/SearchPagingSource.kt:1]: I'll add some explanations along with the documentation next week. It is used in `LibraryRemoteDataSourceImpl.search`

**Pull request merged**